### PR TITLE
log method added

### PR DESCRIPTION
**Pull request by Navid-Ebadi-2003**
log method added @Retriever5500 @MMDPROJECT

### CODE REVIEW

File: src/main/java/Server/Service.java [hunk 1]
```diff
@@ -46,6 +50,21 @@ public void run() {
         }
     }
 
+    public void log (String requestType , UUID userId) {
+        try {
+            if (!userId.equals(null)) {
```
Comment (Retriever5500) [inline src/main/java/Server/Service.java:58]: Here I meant print the log with the userId if it's not null, but if it is just don't include it in the String.

File: src/main/java/Server/Service.java [hunk 2]
```diff
@@ -46,6 +50,21 @@ public void run() {
...
+                FileWriter writer = new FileWriter("log.txt");
+                LocalTime time = LocalTime.now();
+
+                writer.write(userId + "\n" + requestType + "\n" + time + "\n" + "\n");
```
Comment (Retriever5500) [inline src/main/java/Server/Service.java]: Print the time first, it's usually what others do. Also maybe include the client IP address

File: src/main/java/Server/Service.java [hunk 3]
```diff
@@ -72,25 +91,34 @@ public void executeRequest(JsonObject jsonRequest) {
                 String email = requestBody.get("email").getAsString();
                 String username = requestBody.get("username").getAsString();
                 String password = requestBody.get("password").getAsString();
+
+                this.userId = userId;
```
Comment (Retriever5500) [inline src/main/java/Server/Service.java]: Don't set the userId here, maybe the user just signs up and doesn't login. It becomes unnecessary. In `log' pass userId directly instead.

**Review by Retriever5500 — COMMENTED** [review]
LGTM, just some issues I outlined in review.

**Comment by Navid-Ebadi-2003** [comment]
> LGTM, just some issues I outlined in review.

i saw the problems . i will fix them

**Comment by Navid-Ebadi-2003** [comment]
> > LGTM, just some issues I outlined in review.
> 
> i saw the problems . i will fix them

if every thing okay merge it

**Pull request merged**